ravenframework/Optimizers/GeneticAlgorithm.py

```diff
@@ -776,18 +787,49 @@ def _useRealization(self, info, rlz):
 
       # 2 @ n: Crossover from set of parents
       # Create childrenCoordinates (x1,...,xM)
+      # if crossover probability is a float, keep it as is. But, If it's a string, called appropriate function.
+    if(self._crossoverProbType== "static"):
       childrenXover = self._crossoverInstance(parents=parents,
                                               variables=list(self.toBeSampled),
                                               crossoverProb=self._crossoverProb,
                                               points=self._crossoverPoints)
-
+    elif(self._crossoverProb == "linear"):
+        self._crossoverProb = adaptiveLinearCrossoverProbability(self.getIteration(traj),self.limit)
+        childrenXover = self._crossoverInstance(parents=parents,
+                                              variables=list(self.toBeSampled),
+                                              crossoverProb=self._crossoverProb,
+                                              points=self._crossoverPoints)
+        self._crossoverProb = "linear"
+    elif(self._crossoverProb == "quadratic"):
+        self._crossoverProb = adaptiveQuadraticCrossoverProbability(self.getIteration(traj),self.limit)
+        childrenXover = self._crossoverInstance(parents=parents,
+                                              variables=list(self.toBeSampled),
+                                              crossoverProb=self._crossoverProb,
+                                              points=self._crossoverPoints)
+        self._crossoverProb = "quadratic"
```

Comment:
same here:
`crossoverProb = adaptiveQuadraticCrossoverProbability(self.getIteration(traj),self.limit)`
and remove 809

Reply:
After that do you see that 792, 798, and 805 are exactly similar? Doesn't that mean we can take it out of the if statements? Please do that.